### DISCUSSION TITLE
Adobe PDF Embed API deprecation

### DISCUSSION THREAD

**Discussion by clementepereyra** [adobe/aem-core-wcm-components#2373] [General]
Hi all,
This week Adobe informed about the PDF Embed API, "We’ve launched a new modern PDF viewer that will replace the legacy viewer..." "...Please transition over to the new modern PDF viewer because the **legacy viewer will be deprecated on Jan 31, 2023.**"
Currently, the [PDF Viewer component is using](https://github.com/adobe/aem-core-wcm-components/blob/main/content/src/content/jcr_root/apps/core/wcm/components/pdfviewer/v1/pdfviewer/clientlibs/site/js/pdfviewer.js) the legacy version of the API (the [main.js](https://documentservices.adobe.com/view-sdk/main.js) script).
I was wondering if there are plans to release a new version of AEM Core Components including the PDF Viewer upgrade to use the new library (the [viewer.js](https://documentservices.adobe.com/view-sdk/viewer.js) script).

More information can be found here: https://developer.adobe.com/document-services/docs/overview/pdf-embed-api/modernviewer/

**Comment by clementepereyra** [comment]
For those interested in this topic, I have been able to get a response from Adobe through a Support Ticket, since unfortunately this project seems to remain unattended by them.

### Response from Adobe rep
> As per the Adobe Document Cloud Team, the PDF Viewer Core Component is currently based on the deprecated main.js script, while the Modern Viewer is on viewer.js.
> Post-January, the main.js script will still stay in the service but it will be updated to automatically start displaying the Modern Viewer to all the existing users. Clients will be notified of this auto-migration plan.
> Hence, the Customer Experience will continue to work post-January.
> Meanwhile, the AEM Product Engineering Team shall also be trying to release a new version of the Core Component with the new Viewer in the January release of our Core Component.

### TL;DR:
No action is required unless you have customised the PDF Viewer JavaScript.

**Reply by kwin** [reply]
This is tackled in https://github.com/adobe/aem-core-wcm-components/pull/2402 now.
